core/src/banking_stage/transaction_scheduler/transaction_state.rs

```diff
@@ -1,53 +1,32 @@
 use crate::banking_stage::scheduler_messages::MaxAge;
 
-/// Simple wrapper type to tie a sanitized transaction to max age slot.
-pub(crate) struct SanitizedTransactionTTL<Tx> {
-    pub(crate) transaction: Tx,
-    pub(crate) max_age: MaxAge,
-}
-
 /// TransactionState is used to track the state of a transaction in the transaction scheduler
 /// and banking stage as a whole.
 ///
-/// There are two states a transaction can be in:
-///     1. `Unprocessed` - The transaction is available for scheduling.
-///     2. `Pending` - The transaction is currently scheduled or being processed.
-///
-/// Newly received transactions are initially in the `Unprocessed` state.
-/// When a transaction is scheduled, it is transitioned to the `Pending` state,
-///   using the `transition_to_pending` method.
-/// When a transaction finishes processing it may be retryable. If it is retryable,
-///   the transaction is transitioned back to the `Unprocessed` state using the
-///   `transition_to_unprocessed` method. If it is not retryable, the state should
-///   be dropped.
-///
-/// For performance, when a transaction is transitioned to the `Pending` state, the
-///   internal `SanitizedTransaction` is moved out of the `TransactionState` and sent
-///   to the appropriate thread for processing. This is done to avoid cloning the
-///  `SanitizedTransaction`.
-#[allow(clippy::large_enum_variant)]
-pub(crate) enum TransactionState<Tx> {
-    /// The transaction is available for scheduling.
-    Unprocessed {
-        transaction_ttl: SanitizedTransactionTTL<Tx>,
-        priority: u64,
-        cost: u64,
-    },
-    /// The transaction is currently scheduled or being processed.
-    Pending { priority: u64, cost: u64 },
-    /// Only used during transition.
-    Transitioning,
+/// Newly received transactions initially have `Some(transaction)`.
+/// When a transaction is scheduled, the transaction is taken from the Option.
+/// When a transaction finishes processing it may be retryable. If it is
+/// retryable, the transaction is added back into the Option. If it si not
+/// retryable, the state is dropped.
+pub(crate) struct TransactionState<Tx> {
+    /// If `Some`, the transaction is available for scheduling.
+    /// If `None`, the transaction is currently scheduled or being processed.
+    transaction: Option<Tx>,
+    /// Tracks information on the maximum age the transaction's pre-processing
+    /// is valid for.
+    max_age: MaxAge,
+    /// Priority of the transaction.
+    priority: u64,
+    /// Estimated cost of the transaction.
+    cost: u64,
```

Comment:
looks like `max_age`, `priority` and `cost` are properties associated with `transaction` that's ready for processing. If so, they can be bundled together to better describe "state", could also make `container.insert_new_transaction( tx_with_its_properties )` cleaner. wdyt something like this instead of `option<Tx>`?

```
enum TransactionState<Tx> {
    Unprocessed {
        transaction: Tx,
        max_age: MaxAge,
        priority: u64,
        cost: u64,
        // or define a struct for above 4 fields
    },
    Processing,
}
```

Reply:
Think it's fine to make a bundle of these, but only few actually need to be sent with the other items for processing (only `max_age` currently).

Reply:
looks like `TransactionSchedulingInfo` takes both `max_age` and `cost`; `max_age` is to validate `cost` and `priority`, they might fit for bundling. I don't really feel strong either way, can keep it as is, since `TransactionState` is internal to Scheduler.

Reply:
`max_age` is not to validate cost and priority. it's to validate the transaction

Reply:
Oh, the comment for `max_age` lead me to think it is to validate if priority and cost need to be re-compute.

Reply:
ah, i can add some clarification - it's mainly the ALTs that can be closed / expired.

Reply:
a866f9e139